### PR TITLE
[Math-Checker] Add formular comparision

### LINKED ISSUE #986: [Math-Checker] Formular Comparision
**Is your feature request related to a problem? Please describe.**
Currently the math checker can only compare formulas as strings. This lead to mathematically equal formulas being seen as wrong (`a+b` and `b+a`).

**Describe the solution you'd like**
The Math Checker should see Mathematically equal formulas as equal.

### PR DESCRIPTION
**Pull request by Zitrone44**
closes #986

### CODE REVIEW

**Review by mxsph — CHANGES_REQUESTED** [review]

**Review by mxsph — CHANGES_REQUESTED** [review]
- [x] The coverage of the Math-Parser should be ~100% (see [coverage](https://app.codecov.io/gh/thm-mni-ii/feedbacksystem/tree/i986-formular-comparision/modules/fbs-core))

**Comment by Zitrone44** [comment]
> The [coverage](https://app.codecov.io/gh/thm-mni-ii/feedbacksystem/tree/i986-formular-comparision/modules/fbs-core) of the Math-Parser should be ~100% (see coverage)

This can not be archived as their are some lines which are in theory unreachable but are still required to keep the kotline compiler happy.

File: modules/fbs-core/math-parser/src/main/antlr/de/thm/ii/fbs/mathParser/Math.g4 [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+VAR: [a-z];
+
+ADD: '+';
+SUB: '-';
+MUL: '*';
+DIV: '/';
+EXP: '^';
+SQR: 'sqrt';
+LB: 'lb';
```
Comment (pdrtzk) [inline modules/fbs-core/math-parser/src/main/antlr/de/thm/ii/fbs/mathParser/Math.g4:27]: Use _ld_ (logarithmus dualis) as alias for _lb_

File: modules/fbs-core/math-parser/src/main/antlr/de/thm/ii/fbs/mathParser/Math.g4 [hunk 2]
```diff
@@ -0,0 +1,31 @@
+grammar Math;
+expr:   (SQR|LB|LN|LG) ' '* expr
```
Comment (pdrtzk) [inline modules/fbs-core/math-parser/src/main/antlr/de/thm/ii/fbs/mathParser/Math.g4:2]: What about a unary minus (like **-7** * 3)?

**Review by pdrtzk — CHANGES_REQUESTED** [review]

**Review by pdrtzk — DISMISSED** [review]

**Review by mxsph — APPROVED** [review]

**Pull request merged**